src/test/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkTaskWithSchemaTest.java

```diff
@@ -119,6 +119,22 @@ public void mapTypesTest() {
         assertEquals(sr.size(), ClickHouseTestHelpers.countRows(chc, topic));
     }
 
+    @Test
+    public void nullableArrayTypeTest() {
+        Map<String, String> props = getTestProperties();
+        ClickHouseHelperClient chc = createClient(props);
+
+        String topic = "nullable_array_string_table_test";
+        ClickHouseTestHelpers.dropTable(chc, topic);
+        ClickHouseTestHelpers.createTable(chc, topic, "CREATE TABLE %s ( `off16` Int16, `arr` Array(String)  ) Engine = MergeTree ORDER BY off16");
+        Collection<SinkRecord> sr = SchemaTestData.createNullableArrayType(topic, 1);
+
+        ClickHouseSinkTask chst = new ClickHouseSinkTask();
+        chst.start(props);
+        assertThrowsExactly(RuntimeException.class, () -> chst.put(sr), "An attempt to write null into not nullable column 'arr'");
```

Comment:
It should throw exception when null value come to array type, because in this way it cant be resolved.
It should pass empty array! 
In ClickHouse null value equal empty array for Array type.
```
with '{"x": null}' as json
select JSONExtractArrayRaw(json, 'x');
-----
result []
```
<img width="423" alt="image" src="https://github.com/ClickHouse/clickhouse-kafka-connect/assets/14272376/7473acbe-66a6-417e-9a6e-557fb7d401f6">

Reply:
Yeah, this is a small PR that just fixes current behaviour. So instead of Null Pointer Exception something more meaningful will appear in the log. 

I guess we gonna need to wait till maintainers decided what to do with the bug you've submitted, cuz I don't feel comfortable making such changes.